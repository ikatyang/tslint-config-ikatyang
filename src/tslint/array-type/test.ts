// good

type good1 = string[];
type good2 = Array<(x: any) => any>;

// bad

type bad2 = Array<string[]>;
type bad1 = ((x: any) => any)[];
