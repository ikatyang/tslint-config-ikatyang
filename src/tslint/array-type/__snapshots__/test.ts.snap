// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
// good

type good1 = string[];
type good2 = Array<(x: any) => any>;

// bad

type bad2 = Array<string[]>;
            ~~~~~~~~~~~~~~~  [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
type bad1 = ((x: any) => any)[];
            ~~~~~~~~~~~~~~~~~~~  [Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.]

`;
